stl/inc/ranges

```diff
@@ -18,6 +18,9 @@
 #include <span>
 #include <string_view>
 #include <tuple>
+#if _HAS_CXX23
+#include <variant>

```

Comment:
We could implement a simplified `std::variant`. It would be a bit more complicated than `_Variantish` because it needs to support `emplace`, `visit` and `operator==`, but that's tolerable.

But consider that in the future we will have [`views::concat`](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2542r0.html), which needs a variant with an arbitrary number of alternatives. This will require most functionality of `std::variant`.

So in the future (probably C++26) we need to choose between

1. reimplement most of `std::variant`
2. use `std::variant` (and include `<variant>`)

If we choose 2 in C++26, then we might as well use `std::variant` in C++23.

Reply:
I do not think this is something we need to choose later. Even if at some point we are forced to include variant in full, our implementation of `_Variantish` would still dramatically improve throughput compared to using a full variant. So I do not see a scenario, where we would build it back.

I would vote for expanding the current `_Variantish` to also feature `emplace`, `visit` and `operator==`.  That would be cheap.

Reply:
We did switch to `_Variantish` so the only thing we need from varia t is `_Throw_bad_variant_access` should we pull that out to `xutility`

Reply:
Good catch - I'll move it to `<exception>`, which both `<ranges>` and `<variant>` drag in (and which actually mentions the `exception` type, unlike `<xutility>`).